src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartControllerTest.java

```diff
@@ -0,0 +1,145 @@
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.sandl.snlevents.mappers.FactsMapper;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.HearingPart;
+import uk.gov.hmcts.reform.sandl.snlevents.model.request.CreateHearingPart;
+import uk.gov.hmcts.reform.sandl.snlevents.service.HearingPartService;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(HearingPartController.class)
+public class HearingPartControllerTest {
+    public static final String TYPE = "type";
+    public static final String CASE_NUMBER = "90";
+    public static final String TITLE = "title";
+    public static final String HEARING_TYPE = "hearing-type";
+    public static final String URL = "/hearing-part";
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private HearingPartService hearingPartService;
+
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private RulesService rulesService;
+
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private FactsMapper factsMapper;
+
+    private JacksonTester<CreateHearingPart> createHearingPartSerializer;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Before
+    public void setup() {
+        JacksonTester.initFields(this, objectMapper);
+    }
+
+    @Test
+    public void fetchAllHeartingParts_returnHearingPartsFromService() throws Exception {
+        val hearingParts = createHearingParts();
+        when(hearingPartService.getAllHearingParts()).thenReturn(hearingParts);
+
+        val response = mvc
+            .perform(get(URL))
+            .andExpect(status().isOk())
+            .andReturn().getResponse();
+
+        val r = objectMapper.readValue(response.getContentAsString(), new TypeReference<List<HearingPart>>(){});
+        assertThat(r).isEqualToComparingFieldByFieldRecursively(hearingParts);
+    }
+
+    private List<HearingPart> createHearingParts() {
+        return new ArrayList<>(Arrays.asList(createHearingPart()));
```

Comment:
Arrays.asList... is enough, you don't need the new Arraylist<>
In fact you could use here 'Collections.singletonList(createHearingPart());' as intelliJ is suggesting but that's no necessary.

Reply:
Thank you very very, done.